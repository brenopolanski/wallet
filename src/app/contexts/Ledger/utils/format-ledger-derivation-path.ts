import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
