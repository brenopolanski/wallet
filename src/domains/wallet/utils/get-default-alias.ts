import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
