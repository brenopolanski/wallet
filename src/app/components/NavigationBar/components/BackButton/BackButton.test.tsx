import { createMemoryHistory } from "history";
import React from "react";
import { fireEvent, render, screen } from "utils/testing-library";

import { BackButton } from "./BackButton";
const history = createMemoryHistory();

describe("BackButton", () => {
	it("should render", () => {
		const { container } = render(<BackButton />, { history });

		expect(container).toHaveTextContent("chevron-left-small.svg");
		expect(container).toMatchSnapshot();
	});

	it("should render when disabled", () => {
		const { container } = render(<BackButton disabled />, { history });

		expect(container).toHaveTextContent("chevron-left-small.svg");
		expect(container).toMatchSnapshot();
	});

	it("should redirect to previous page", () => {
		const historySpy = jest.spyOn(history, "go").mockImplementation();

		const { container } = render(<BackButton />, { history });

		fireEvent.click(screen.getByRole("button"));

		expect(historySpy).toHaveBeenCalledWith(-1);

		expect(container).toHaveTextContent("chevron-left-small.svg");
		expect(container).toMatchSnapshot();

		historySpy.mockRestore();
	});

	it("should redirect to given url", () => {
		const historySpy = jest.spyOn(history, "push").mockImplementation();

		const { container } = render(<BackButton backToUrl="new-url" />, { history });

		fireEvent.click(screen.getByRole("button"));

		expect(historySpy).toHaveBeenCalledWith("new-url");

		expect(container).toHaveTextContent("chevron-left-small.svg");
		expect(container).toMatchSnapshot();

		historySpy.mockRestore();
	});

	it("should not redirect to previous page when disabled", () => {
		const historySpy = jest.spyOn(history, "go").mockImplementation();

		const { container } = render(<BackButton disabled />, { history });

		fireEvent.click(screen.getByRole("button"));

		expect(historySpy).not.toHaveBeenCalled();

		expect(container).toHaveTextContent("chevron-left-small.svg");
		expect(container).toMatchSnapshot();

		historySpy.mockRestore();
	});

	it("should not redirect to given url when disabled", () => {
		const historySpy = jest.spyOn(history, "push").mockImplementation();

		const { container } = render(<BackButton backToUrl="new-url" disabled />, { history });

		fireEvent.click(screen.getByRole("button"));

		expect(historySpy).not.toHaveBeenCalled();

		expect(container).toHaveTextContent("chevron-left-small.svg");
		expect(container).toMatchSnapshot();

		historySpy.mockRestore();
	});
});
