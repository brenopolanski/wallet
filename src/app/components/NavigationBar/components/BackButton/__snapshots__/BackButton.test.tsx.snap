// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton should not redirect to given url when disabled 1`] = `
<div>
  <button
    class="BackButton__StyledBackButton-sc-18ta3r7-0 iaWPTU"
    disabled=""
  >
    <div
      class="sc-bdvvtL Cfvxc mx-auto"
      height="10"
      width="10"
    >
      <svg>
        chevron-left-small.svg
      </svg>
    </div>
  </button>
</div>
`;

exports[`BackButton should not redirect to previous page when disabled 1`] = `
<div>
  <button
    class="BackButton__StyledBackButton-sc-18ta3r7-0 iaWPTU"
    disabled=""
  >
    <div
      class="sc-bdvvtL Cfvxc mx-auto"
      height="10"
      width="10"
    >
      <svg>
        chevron-left-small.svg
      </svg>
    </div>
  </button>
</div>
`;

exports[`BackButton should redirect to given url 1`] = `
<div>
  <button
    class="BackButton__StyledBackButton-sc-18ta3r7-0 gAKHms"
  >
    <div
      class="sc-bdvvtL Cfvxc mx-auto"
      height="10"
      width="10"
    >
      <svg>
        chevron-left-small.svg
      </svg>
    </div>
  </button>
</div>
`;

exports[`BackButton should redirect to previous page 1`] = `
<div>
  <button
    class="BackButton__StyledBackButton-sc-18ta3r7-0 gAKHms"
  >
    <div
      class="sc-bdvvtL Cfvxc mx-auto"
      height="10"
      width="10"
    >
      <svg>
        chevron-left-small.svg
      </svg>
    </div>
  </button>
</div>
`;

exports[`BackButton should render 1`] = `
<div>
  <button
    class="BackButton__StyledBackButton-sc-18ta3r7-0 gAKHms"
  >
    <div
      class="sc-bdvvtL Cfvxc mx-auto"
      height="10"
      width="10"
    >
      <svg>
        chevron-left-small.svg
      </svg>
    </div>
  </button>
</div>
`;

exports[`BackButton should render when disabled 1`] = `
<div>
  <button
    class="BackButton__StyledBackButton-sc-18ta3r7-0 iaWPTU"
    disabled=""
  >
    <div
      class="sc-bdvvtL Cfvxc mx-auto"
      height="10"
      width="10"
    >
      <svg>
        chevron-left-small.svg
      </svg>
    </div>
  </button>
</div>
`;
